### PR TITLE
Lua autocmds

### PR DESCRIPTION
**Pull request by rockerBOO**
This replaces autocmds with lua autocmds. We may want to support both until 0.7 is released with hopefully lua autocmds. 

Requires neovim/neovim#14661

### CODE REVIEW

**Comment by Conni2461** [comment]
POGSLIDE

but we can only merge it if its in an official release. And that release is the only release we support.

**Comment by rockerBOO** [comment]
Updated to official API release.

**Comment by rockerBOO** [comment]
`:Telescope git_files` closes immediately so I feel something is wrong with my implementation. Will need to test more and take a look.

File: lua/telescope/actions/init.lua [hunk 1]
```diff
@@ -1010,14 +1010,14 @@ actions.which_key = function(prompt_bufnr, opts)
...
-  vim.cmd(string.format(
-    "autocmd BufLeave <buffer> ++once lua %s",
-    table.concat({
-      string.format("pcall(vim.api.nvim_win_close, %s, true)", km_win_id),
-      string.format("pcall(vim.api.nvim_win_close, %s, true)", km_opts.border.win_id),
-      string.format("require 'telescope.utils'.buf_delete(%s)", km_buf),
-    }, ";")
-  ))
+  vim.api.nvim_create_autocmd("BufLeave", {
+    once = true,
```
Comment (tjdevries) [inline lua/telescope/actions/init.lua:1014]: you missed making this a buffer local autocmd

File: lua/telescope/pickers.lua [hunk 2]
```diff
@@ -498,22 +498,24 @@ function Picker:find()
...
-    prompt_bufnr
-  )
-
-  vim.cmd [[augroup PickerInsert]]
-  vim.cmd [[  au!]]
-  vim.cmd(on_buf_leave)
-  vim.cmd(on_vim_resize)
-  vim.cmd [[augroup END]]
+  vim.api.nvim_create_autocmd("BufLeave", {
+    group = "PickerInsert",
```
Comment (tjdevries) [inline lua/telescope/pickers.lua:504]: same w/ these

**Comment by rockerBOO** [comment]
Thanks teej!

Added the buffer to the autocmds and this resolved the live_grep and other issues I was having (like with git). Will use as my main branch to do more testing but seems good so far.

**Review by Conni2461 — APPROVED** [review]
Thanks rocker :) I just rebased and tested it  and everything seems to work

I intend to merge this PR on Friday after https://github.com/nvim-telescope/telescope.nvim/pull/1851

**Comment by Conni2461** [comment]
Lets go lua autocmds.

Thanks again :)

**Pull request merged**